File: tests/human/x/zig/record_assign.zig
const std = @import("std");

const Counter = struct { n: i32 };

fn inc(c: *Counter) void {
    c.n += 1;
}

pub fn main() void {
    var c = Counter{ .n = 0 };
    inc(&c);
    std.debug.print("{d}\n", .{c.n});
}

File: tests/human/x/scala/record_assign.scala
case class Counter(var n: Int)

object record_assign {
  def inc(c: Counter): Unit = {
    c.n = c.n + 1
  }
  def main(args: Array[String]): Unit = {
    val c = Counter(0)
    inc(c)
    println(c.n)
  }
}
